src/dotty/tools/dotc/core/Types.scala

```diff
@@ -810,6 +810,13 @@ object Types {
       case _ => this
     }
 
+    /** Eliminate anonymous classes */
+    final def elimAnonymousClass(implicit ctx: Context): Type = this match {
```

Comment:
`deAnonymize` seems more in sync with `deAlias`.
Additionally  `elim` at least in my head means eliminating deeply inside the type, ie if there's a reference to an anonymous class somewhere deep you'll also eliminate it.